com.reprezen.swagedit/src/com/reprezen/swagedit/model/AbstractNode.java

```diff
@@ -175,7 +176,11 @@ public Position getPosition(IDocument document) {
         int endCol = getEnd().getColumn();
         try {
             offset = document.getLineOffset(startLine);
-            length = (document.getLineOffset(endLine) + endCol) - offset;
+            if (selectEntireElement) {
+                length = (document.getLineOffset(endLine) + endCol) - offset;
+            } else {
+                length = document.getLineOffset(startLine+1) - offset;
```

Comment:
Coulld this throw BadLocationException if startLine is the last line in the document?